x/ccv/provider/keeper/proposal.go

```diff
@@ -218,6 +218,15 @@ func (k Keeper) StopConsumerChain(ctx sdk.Context, chainID string, lockUbd, clos
 		k.DeleteUnbondingOpIndex(ctx, chainID, id)
 	}
 
+	// Remove any existing throttling related queue entries from the
+	// global queue for this consumer.
+	k.DeletePendingSlashPacketEntriesForConsumer(ctx, chainID)
+
+	// Remove all pending slash packets and vsc matured packets queued for this consumer.
+	// Note: queued VSC matured packets can be safely removed from the per-chain queue,
```

Comment:
Could someone confirm for me that it is indeed safe to throw out queued VSC matured packets when stopping a consumer? This seems safe since the unbonding ops are already released above

Reply:
Seems good to me. @mpoke?

Reply:
Removing the queued VSCMatured packets is fine. I don't like the idea that we remove queued slash packets. These packets were already relayed on the provider and the only reason were not handled is the throttling mechanism. IMO, a consumer state should not be removed until the queue of received packets is empty.